### PR TITLE
convert `sub` to TypeScript

### PR DESCRIPTION
**Pull request by fturmel**
Older PR for reference #2118

### CODE REVIEW

**Comment by tan75** [comment]
This PR closes https://github.com/date-fns/date-fns-typescriptify/issues/138

**Review by tan75 — APPROVED** [review]
Thank you for this PR! 🙏

File: src/sub/index.ts [hunk 1]
```diff
@@ -43,7 +43,10 @@ import toInteger from '../_lib/toInteger/index'
  * })
  * //=> Mon Sep 1 2014 10:19:50
  */
-export default function sub(dirtyDate, duration) {
+export default function sub(
+  dirtyDate: Date | number,
```
Comment (kossnocorp) [inline src/sub/index.ts:47]:
```suggestion
  date: Date | number,
```

File: src/sub/index.ts [hunk 2]
```diff
@@ -57,7 +60,7 @@ export default function sub(dirtyDate, duration) {
   const seconds = duration.seconds ? toInteger(duration.seconds) : 0
 
   // Subtract years and months
-  const dateWithoutMonths = subMonths(toDate(dirtyDate), months + years * 12)
+  const dateWithoutMonths = subMonths(dirtyDate, months + years * 12)
```
Comment (kossnocorp) [inline src/sub/index.ts:63]:
```suggestion
  const dateWithoutMonths = subMonths(date, months + years * 12)
```

**Review by kossnocorp — APPROVED** [review]

**Pull request merged**